### PR TITLE
fix: zsh inplace activation

### LINKED ISSUE #747: activate: inplace activation does not run profile.d scripts for zsh
ZSH activation runs profile.d scripts exclusively from the .zshrc files.
For inplace activation they should be run as well (like the bash flow)

### PR DESCRIPTION
**Pull request by ysndr**
* add a behavioral test
* fix zsh activation - profile.d scripts were not run when activating inplace

Follow-up on #735
fixes #747

### CODE REVIEW

File: cli/tests/environment-activate.bats [hunk 1]
```diff
@@ -375,3 +375,30 @@ env_is_activated() {
   assert_success
   assert_output --regexp "source .*/activate/zsh"
 }
+
+# bats test_tags=activate,activate:inplace-modifies
+@test "'flox activate' allows to modify current shell" {
```
Comment (mkenigs) [inline cli/tests/environment-activate.bats:380]: nit
```suggestion
@test "'flox activate' modifies the current shell" {
```

Comment (mkenigs) [inline cli/tests/environment-activate.bats:380]: nit: other tests in the file are split up a bit more granularly. I don't particularly care but I thought you and @zmitchell sometimes prefer that 🤷

**Review by mkenigs — APPROVED** [review]
This is why we write tests hehe

File: cli/tests/environment-activate.bats (hunk 1 above)
Comment (ysndr) [reply cli/tests/environment-activate.bats]: split them up...

**Review by mkenigs — APPROVED** [review]

**Pull request merged**